api/campaigns/class-maybe-show-campaign.php

```diff
@@ -132,24 +189,22 @@ function ( $post_data ) use ( $visit ) {
 	 * @param string $page_referer_url URL of the referrer of the frontend page that is making the API request.
 	 * @param object $view_as_spec "View As" specification.
 	 * @param string $now Current timestamp.
+	 * @param bool   $check_segment Whether to check if the client matches the prompt's segment. If false, the client is assumed to match the segment.
```

Comment:
I may be missing something here, but `should_campaign_be_shown` is called only [here](https://github.com/Automattic/newspack-popups/blob/9609176353d81f454cc4f781548844dd744e5825/api/campaigns/class-maybe-show-campaign.php#L103), with `$check_segment` of value `false`. Therefore, why is this argument needed?

Reply:
This was maybe a cheating way to get some failing tests to pass. In this branch, we check whether the client matches each segment, then only run `should_campaign_be_shown` on the campaigns with a matching segment. That caused many tests to fail unless we re-checked whether the client matches the segment in `should_campaign_be_shown`.

In `feat/multiple-segments`, I've gone back to running `should_campaign_be_shown` on every campaign, not just ones that match the segment, so I was able to remove the argument here: https://github.com/Automattic/newspack-popups/pull/451/files#diff-fab5ae21610a1add85a1640bebc4871f8fa855dd8f7c88d149e2f1c4c20df540R196

Reply:
Since this is fixed but with a lot of logical reshuffling in #451, I suggest we leave it for this branch since it doesn't cause any problems and will be overwritten immediately by merging #451.